packages/container/lib/container.js

```diff
@@ -289,16 +291,23 @@ if (isFeatureEnabled('ember-factory-for')) {
       if (!normalizedName) { return; }
     }
 
+    let cached = this.factoryManagerCache[normalizedName];
+
+    if (cached) { return cached; }
+
     let factory = this.registry.resolve(normalizedName);
 
-    if (factory === undefined) { return; }
+    if (factory === undefined) {
```

Comment:
i thought so, but there where test failures and I'm tired :P Maybe i can convince  @chadhietala to investigate?